modules/serialize/src/identifiers.ts

```diff
@@ -62,8 +62,34 @@ Object.freeze(SequenceIdentifier)
 export enum Maximum {
   // Maximum number of messages which are allowed to be encrypted under a single cached data key
   MESSAGES_PER_KEY = 2 ** 32, // eslint-disable-line no-unused-vars
-  // Maximum number of bytes which are allowed to be encrypted under a single cached data key
-  BYTES_PER_KEY = 2 ** 63 - 1, // eslint-disable-line no-unused-vars
+  /* Maximum number of bytes which are allowed to be encrypted
+   * under a cached single data key across messages.
+   * The _real_ maximum is 2 ** 63 - 1,
```

Comment:
I still think this is misleading; we're giving too little information for this to actually be accurate.

My gut is that we should probably not talk a lot about the *why* in a comment in a source file; that is better served by proper documentation in our concepts overviews.

There are three value at play here:

**Cryptographic Safety**

Maximum number of bytes that we will allow within a single AES operation: 2^63 - 32.
We chose this limit based on the mathematical characteristics of the AES algorithm.
This is also the maximum number of bytes that we allow in an unframed message because the entire body is a single AES operation.

**Message Format Bounds**

Maximum number of bytes that we will allow in a single frame: 2^32 - 1.
This limit is defined and enforced by the message format.

Maximum frames that we will allow in a single framed message: 2^32 - 1.
This limit is defined and enforced by the message format.

Maximum number of bytes we will allow in a single framed message: (2^32 - 1) * (2^32 - 1).
This limit is a consequence of the previous two.

**Caching Constraints**

Maximum number of messages we will allow to be protected under a single data key: 2^32.

Maximum number of bytes we will allow to be protected under a single data key before we stop reusing that data key: 2^63 - 1.
* As you pointed out, this is not exactly a limit on the number of bytes protected; it is more correctly a point at which we will cease reuse.